src/Symfony/Component/Form/Extension/Validator/Constraints/FormValidator.php

```diff
@@ -90,7 +91,9 @@ public function validate($form, Constraint $formConstraint)
                             $validator->atPath('data')->validate($form->getData(), $constraint, $group);
 
                             // Prevent duplicate validation
-                            continue 2;
+                            if (!$constraint instanceof Composite) {
```

Comment:
This is basically the same fix we already applied in the Validator component last year in #29499.